python/cugraph/centrality/betweenness_centrality_wrapper.pyx

```diff
@@ -26,82 +27,227 @@ import cudf
+    c_indices = indices.__cuda_array_interface__['data'][0]
+
+    number_of_vertices = len(offsets) - 1
+    number_of_edges = len(indices)
+
+    result_size = number_of_vertices
+    result_df = get_output_df(result_size, result_dtype)
+    number_of_sources_in_batch = len(batch)
+    if result_dtype == np.float64:
+        graph_double = GraphCSRView[int, int, double](<int*> c_offsets,
+                                                      <int*> c_indices,
+                                                      <double*> c_graph_weights,
+                                                      number_of_vertices,
+                                                      number_of_edges)
+        graph_double.prop.directed = is_directed
+        c_graph = <uintptr_t>&graph_double
+    elif result_dtype == np.float32:
+        graph_float = GraphCSRView[int, int, float](<int*>c_offsets,
+                                                    <int*>c_indices,
+                                                    <float*>c_graph_weights,
+                                                    number_of_vertices,
+                                                    number_of_edges)
+        graph_float.prop.directed = is_directed
+        c_graph = <uintptr_t>&graph_float
+    else:
+        raise ValueError("result_dtype can only be np.float64 or np.float32")
+
+    c_identifier = result_df['vertex'].__cuda_array_interface__['data'][0]
+    c_betweenness = result_df['betweenness_centrality'].__cuda_array_interface__['data'][0]
+    if weights is not None:
+        c_weights = weights.__cuda_array_interface__['data'][0]
+
+    c_batch = batch.__array_interface__['data'][0]
+    c_handle = <uintptr_t>handle.getHandle()
+
+    run_c_betweenness_centrality(c_handle,
+                                 c_graph,
+                                 c_betweenness,
+                                 normalized,
+                                 endpoints,
+                                 c_weights,
+                                 number_of_sources_in_batch,
+                                 c_batch,
+                                 total_number_of_sources,
+                                 result_dtype)
+    if result_dtype == np.float64:
         graph_double.get_vertex_identifiers(<int*> c_identifier)
+    elif result_dtype == np.float32:
+        graph_float.get_vertex_identifiers(<int*> c_identifier)
     else:
-        raise TypeError("result type for betweenness centrality can only be "
-                        "float or double")
+        raise ValueError("result_dtype can only be np.float64 or np.float32")
+
+    return result_df
+
+def run_mg_work(input_data, normalized, endpoints,
+                weights, sources,
+                result_dtype, session_id):
+    result = None
+    # 1. Get session information
+    session_state = worker_state(session_id)
+    number_of_workers = session_state["nworkers"]
+    worker_idx = session_state["wid"]
```

Comment:
these can be accessed as

cugraph/python/cugraph/dask/link_analysis/pagerank.py

Line 24 in 38b032e

 wid = Comms.get_worker_id(sID) 
instead of raft import